### PR TITLE
Update cancellation reasons for SupporterPlus and Recurring contributions

### PR DESCRIPTION
**Pull request by rBangay**
### What does this PR change?
Updates the cancellation reasons for supporter plus and recurring contributions. Consolidates the list of reasons and add's 2 new options.

See the raw data in "Cancellation journeys" doc https://docs.google.com/spreadsheets/d/1rf0M-TYmXltJkLHUy7hIeVtezQMmfg4TOaJfMXD3ltk/edit?gid=1054852399#gid=1054852399

### images

Contribution Before             |  Contribution After
:-------------------------:|:-------------------------:
  |  

Supporter plus Before             |  Supporter plus After
:-------------------------:|:-------------------------:
  |

### CODE REVIEW

**Comment by johnduffell** [comment]
@rBangay just to make it easier to double check - where's the original reference for what the changes are supposed to be
Also a screenshot or two might make it easier to review if you have one handy.

File: client/components/mma/cancel/cancellationReason.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export type CancellationReasonId =
 	| 'mma_cost_of_living'
 	| 'mma_cutting_subscriptions'
 	| 'mma_payment_issue'
+	| 'mma_price_increase'
 	| 'mma_article'
```
Comment (johnduffell) [inline client/components/mma/cancel/cancellationReason.ts:41]: we can remove the unused ones now? e.g. I don't think `mma_article` exists in this repo after this change

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { BreakFromNewsWithAlternative } from '../GenericSaveBodyResponses';
+
+export const tierThreeCancellationReasons: CancellationReason[] = [
+	{
+		reasonId: 'mma_editorial',
+		linkLabel: 'I am unhappy with Guardian journalism',
+		saveBody: [
+			'In order to improve our journalism, we’d love to know more about why you are thinking of cancelling.',
+		],
+		alternateFeedbackIntro: standardAlternateFeedbackIntro,
```
Comment (johnduffell) [inline client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:15]: is this making the "custom feedback form prefix message" of "Please share any further thoughts you have about cancelling — you can help us improve. Thank you."

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import {
+	inOrderToImproveSubs,
+	standardAlternateFeedbackIntro,
+} from '../cancellationConstants';
+import type { CancellationReason } from '../cancellationReason';
+import { BreakFromNewsWithAlternative } from '../GenericSaveBodyResponses';
+
+export const tierThreeCancellationReasons: CancellationReason[] = [
```
Comment (johnduffell) [inline client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:8]: do we need to shuffle these in the same way as the others? and if so, do we need to make sure the mma_other option is always last?

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	{
+		reasonId: 'mma_editorial',
```
Comment (johnduffell) [inline client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:10]: the doc doesn't mention what the reasonIds should be.  These all look reasonable though.

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx (hunk 2 above)
Comment (johnduffell) [reply client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:15]: yes it is
https://github.com/guardian/manage-frontend/blob/16068c6661741d982c140afe39e54711bffe2e92/client/components/mma/cancel/cancellationConstants.ts#L3-L4

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		linkLabel: 'I am taking a break from news',
+		saveBody: BreakFromNewsWithAlternative,
+		escalationSaveBody: [
+			'We’d love to know more about what we could do better to help provide inspiring and trustworthy news.',
+		],
+	},
+	{
+		reasonId: 'mma_values',
+		linkLabel: 'I don’t feel that The Guardian values my support',
+		alternateFeedbackIntro: inOrderToImproveSubs,
```
Comment (johnduffell) [inline client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:57]: looks good
https://github.com/guardian/manage-frontend/blob/16068c6661741d982c140afe39e54711bffe2e92/client/components/mma/cancel/cancellationConstants.ts#L6-L7

**Review by johnduffell — APPROVED** [review]
looks accurate, I've updated the description with the doc link.

File: client/components/mma/cancel/cancellationReason.ts (hunk 1 above)
Comment (rBangay) [reply client/components/mma/cancel/cancellationReason.ts:41]: yep good shout, I'll remove all the unused reasons 👍

File: client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx (hunk 2 above)
Comment (rBangay) [reply client/components/mma/cancel/tierThree/TierThreeCancellationReasons.tsx:15]: yep

File: client/components/mma/cancel/cancellationReason.ts (hunk 1 above)
Comment (rBangay) [reply client/components/mma/cancel/cancellationReason.ts]: yep good shout, I'll remove all the unused reasons.

**Comment by johnduffell** [comment]
small tip - try not to force push once it's been looked at by someone else as it makes it hard to see what's happened since the last review (and I ideally don't want to review the whole thing again)

File: client/utilities/utils.ts [hunk 6]
```diff
@@ -6,6 +6,9 @@ export function flattenEquivalent<T>(x: T): T {
 	return x;
 }
 
+export const shuffleArray = (array: unknown[]) =>
+	[...array].sort(() => 0.5 - Math.random());
```
[the remark this answers is not in the document]
Comment (johnduffell) [reply client/utilities/utils.ts:10]: tried in dev console and it's true, this is the percentage of the time each element appears first
```
Object.entries([...Array(1000000).keys().map(() => [1,2,3,4,5,6,7,8,9,10].sort( () => .5 - Math.random() )[0])].sort().reduce((acc, num) => {const e = {...acc};e[num]=acc[num] + 1; return e;}, {1:0,2:0,3:0,4:0,5:0,6:0,7:0,8:0,9:0,10:0})).map(([num, count]) => `${num}: ${count/10000}%`)
(10) ['1: 19.4903%', '2: 7.2752%', '3: 9.6635%', '4: 12.8796%', '5: 8.3072%', '6: 8.9089%', '7: 9.9296%', '8: 11.1513%', '9: 5.9906%', '10: 6.4038%']
```

you can see they should all be 10% but they are repeatably (very) strange amounts

**Pull request merged**